### ISSUE TITLE
Discrepancy Between LAPA and Scratch Results on Simpler and Paper Report

### ISSUE THREAD

**Issue by bighuang624** [LatentActionPretraining/LAPA#6]
Hello! Thank you for your excellent work and for making the code open source. However, I am facing inconsistencies between the results I obtained when reproducing LAPA on the Simpler simulation environment and those reported in the paper. Considering that the released checkpoint might be based on Open-X rather than Bridgev2 as mentioned in the paper, I also reproduced the Scratch results for verification, that is, I fine-tuned directly on the LWM checkpoint using the Simpler data you released. Unfortunately, this also seems to have discrepancies compared to the paper's report. I would like to know the reason for this. Is it possibly due to the need for multiple random seed experiments (I have only experimented 24 times with the default random seed from the source code)? Alternatively, could you release a test code specifically for Simpler (also, I am eagerly looking forward to the release of the Latent Action Quantization code)?

For reference, here is my experimental log:
**LAPA**:
Task name: widowx_stack_cube
Success num: {'moved_correct_obj': 5, 'moved_wrong_obj': 2, 'is_src_obj_grasped': 8, 'consecutive_grasp': 5, 'src_on_target': 0}

Task name: widowx_carrot_on_plate
Success num: {'moved_correct_obj': 9, 'moved_wrong_obj': 1, 'is_src_obj_grasped': 2, 'consecutive_grasp': 1, 'src_on_target': 0}

Task name: widowx_spoon_on_towel
Success num: {'moved_correct_obj': 7, 'moved_wrong_obj': 3, 'is_src_obj_grasped': 1, 'consecutive_grasp': 1, 'src_on_target': 1}

Task name: widowx_put_eggplant_in_basket
Success num: {'moved_correct_obj': 22, 'moved_wrong_obj': 0, 'is_src_obj_grasped': 18, 'consecutive_grasp': 15, 'src_on_target': 11}

**Scratch**:
Task name: widowx_stack_cube
Success num: {'moved_correct_obj': 3, 'moved_wrong_obj': 0, 'is_src_obj_grasped': 3, 'consecutive_grasp': 1, 'src_on_target': 0}

Task name: widowx_carrot_on_plate
Success num: {'moved_correct_obj': 11, 'moved_wrong_obj': 0, 'is_src_obj_grasped': 2, 'consecutive_grasp': 1, 'src_on_target': 0}

Task name: widowx_spoon_on_towel
Success num: {'moved_correct_obj': 2, 'moved_wrong_obj': 4, 'is_src_obj_grasped': 3, 'consecutive_grasp': 2, 'src_on_target': 0}

Task name: widowx_put_eggplant_in_basket
Success num: {'moved_correct_obj': 19, 'moved_wrong_obj': 0, 'is_src_obj_grasped': 8, 'consecutive_grasp': 8, 'src_on_target': 1}

**Comment by seonghyeonye** [comment]
We will shortly release the test code specifically for Simpler. One change we applied from original Simpler codebase is early stopping during evaluation (terminating the trajectory evaluation when the success is True for the first time rather than going through all max steps). We changed this because during fine-tuning data collection through rollouts, we also used this implementation. Therefore, the trained policy showed some unintended behavior after it has completed the task (since it has not learned how to act after the initial success), leading to performance degradation.

**Comment by bighuang624** [comment]
@seonghyeonye Thanks for you response. I have modified the code according to your suggestion, but the result has not changed. Therefore, a test code that is released as soon as possible would be very helpful. Salute.

**Comment by bighuang624** [comment]
@seonghyeonye @joeljang  Thanks a lot for your updates! However, it seems that the inference script `scripts/lapa_bridge.sh` is missed, as it has been mentioned in the updated Simpler Section in README.

**Comment by bighuang624** [comment]
> @seonghyeonye @joeljang Thanks a lot for your updates! However, it seems that the inference script `scripts/lapa_bridge.sh` is missed, as it has been mentioned in the updated Simpler Section in README.

Sorry, I have found it, `SimplerEnv/scripts/lapa_bridge.sh`

**Comment by sroutray** [comment]
Hi @bighuang624 , were you able to reproduce the result? I tried collecting a new set of 100 successful trajectories from the environment using Octo policy, and fine-tuned the OpenX checkpoint on it. While the training accuracy was really high, when evaluating, it performed poorly. I also added @seonghyeonye suggestion to do early stopping after initial success for both data collection and evaluation.

**Comment by JaehongJaehongMin** [comment]
Hi @bighuang624  @sroutray ,

Thank you for pointing out the fine-tuning issue with SimplerEnv.

I'm currently working on fine-tuning other VLA models using collected rollouts from SimplerEnv.
I conducted two separate training runs — one using raw_action as shown [here](https://github.com/simpler-env/SimplerEnv/blob/main/simpler_env/evaluation/maniskill2_evaluator.py#L112),
and another using actions discretized with pd.qcut. Unfortunately, both versions are showing poor performance.

I’d greatly appreciate any advice or suggestions you might have on this.

Thank you!
